datalad/core/local/configuration.py

```diff
@@ -0,0 +1,337 @@
+    supported:
+
+    - 'dataset': the persistent configuration in .datalad/config of a dataset
+    - 'local': a dataset clone's Git repository configuration in .git/config
+    - 'global': non-dataset-specific configuration (usually in $USER/.gitconfig)
+
+    Modifications of the persistent 'dataset' configuration will not be saved
+    by this command, but have to be committed with a subsequent `save`
+    call.
+
+    Rules of precedence regarding different configuration scopes are the same
+    as in Git, with two exceptions: 1) environment variables can be used to
+    override any datalad configuration, and have precedence over any other
+    configuration scope (see below). 2) the 'dataset' scope is considered in
+    addition to the standard git configuration scopes. Its content has lower
+    precedence than Git configuration scopes, but it is committed to a dataset,
+    hence can be used to ship (default and branch-specific) configuration with
+    a dataset.
+
+    Any DATALAD_* environment variable is also mapped to a configuration item.
+    Their values take precedence over any other specification. In variable
+    names '_' encodes a '.' in the configuration name, and '__' encodes a '-',
+    such that 'DATALAD_SOME__VAR' is mapped to 'datalad.some-var'.
+
+    Recursive operation is supported for querying and modifying configuration
+    across a hierarchy of datasets.
+    """
+    _examples_ = [
+        dict(text="Dump the effective configuration, including an annotation for common items",
+             code_py="configuration()",
+             code_cmd="datalad configuration"),
+        dict(text="Query two configuration items",
+             code_py="configuration('get', ['user.name', 'user.email'])",
+             code_cmd="datalad configuration get user.name user.email"),
+        dict(text="Recursively set configuration in all (sub)dataset repositories",
+             code_py="configuration('set', [('my.config.name', 'value')], recursive=True)",
+             code_cmd="datalad configuration -r set my.config=value"),
+        dict(text="Modify the persistent dataset configuration (changes are not committed)",
+             code_py="configuration('set', [('my.config.name', 'value')], scope='dataset')",
+             code_cmd="datalad configuration --scope dataset set my.config=value"),
+    ]
+
+    result_renderer = 'tailored'
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to query or to configure""",
+            constraints=EnsureDataset() | EnsureNone()),
+        action=Parameter(
+            args=("action",),
+            nargs='?',
+            doc="""which action to perform""",
+            constraints=EnsureChoice(*config_actions)),
+        scope=Parameter(
+            args=("--scope",),
+            doc="""identify to configuration scope for getting or setting
```

Comment:
`s/indentify to/s/indentify/` or `s/identify to //` ?

Reply:
Fixed!